.github/workflows/build.yml

```diff
@@ -0,0 +1,130 @@
+name: Build FreeOrion and run unit tests
+on:
+  pull_request:
+    types:
+      - opened
+      - edited
+      - reopened
+      - synchronize
+  push:
+    branches:
+      - master
```

Comment:
does this mean it will only rub test builds on the master branch when it is pushed an no others unless they are part of a pull request?

Reply:
Yes. Do we need to build pushes always?

Reply:
I find it handy to have a test branch that I can push to test build without needing a pull request always open. Could just add the [refactor_test](https://github.com/freeorion/freeorion/tree/refactor_test) branch for that, though, I suppose.

Reply:
We can add a special pattern for that.  For example `ci_*`

https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags